mlflow_faculty/filter.py

```diff
@@ -0,0 +1,402 @@
+    elif view_type == ViewType.ALL:
+        return None
+    else:
+        raise ValueError("Invalid ViewType: {}".format(view_type))
+
+
+def parse_filter_string(mlflow_filter_string):
+    """Parse an MLflow filter string into a Faculty filter object."""
+    try:
+        parsed = sqlparse.parse(mlflow_filter_string)
+    except Exception:
+        raise ValueError(
+            "Error parsing filter '{}'".format(mlflow_filter_string)
+        )
+
+    try:
+        [statement] = parsed
+    except ValueError:
+        raise ValueError(
+            "Invalid filter '{}'. Must be a single statement.".format(
+                mlflow_filter_string
+            )
+        )
+
+    if not isinstance(statement, SqlStatement):
+        raise ValueError(
+            "Invalid filter '{}'. Must be a single statement.".format(
+                mlflow_filter_string
+            )
+        )
+
+    return _parse_token_list(statement.tokens)
+
+
+def _parse_token_list(tokens):
+    """Parse a list of sqlparse Tokens and return an equivalent filter."""
+
+    # Ignore whitespace chars
+    tokens = [t for t in tokens if not t.is_whitespace]
+
+    if any(_is_or(t) for t in tokens):
+        filters = []
+        for part in _split_list(tokens, _is_or):
+            filters.append(_parse_token_list(part))
+        return CompoundFilter(LogicalOperator.OR, filters)
+
+    elif any(_is_and(t) for t in tokens):
+        filters = []
+        for part in _split_list(tokens, _is_and):
+            filters.append(_parse_token_list(part))
+        return CompoundFilter(LogicalOperator.AND, filters)
+
+    elif len(tokens) == 1:
+        [token] = tokens
+        if isinstance(token, SqlParenthesis):
+            # Strip opening and closing parentheses
+            return _parse_token_list(token.tokens[1:-1])
+        elif isinstance(token, SqlComparison):
+            return _parse_token_list(token.tokens)
+        else:
+            raise ValueError(
+                "Unsupported filter string component: {!r}".format(
+                    token.normalized
+                )
+            )
+
+    elif len(tokens) == 3:
+        return _single_filter_from_tokens(*tokens)
+
+    else:
+        raise ValueError(
+            "Unsupported filter string component: {!r}".format(
+                " ".join(t.normalized for t in tokens)
+            )
+        )
+
+
+def _is_and(token):
+    return token.match(ttype=SqlTokenType.Keyword, values=["AND"])
+
+
+def _is_or(token):
+    return token.match(ttype=SqlTokenType.Keyword, values=["OR"])
+
+
+def _split_list(source_list, condition):
+    chunk = []
+    for value in source_list:
+        if condition(value):
+            yield chunk
+            chunk = []
+        else:
+            chunk.append(value)
+    yield chunk
+
+
+def _single_filter_from_tokens(identifier_token, operator_token, value_token):
+    key_type, key = _parse_identifier(identifier_token)
```

Comment:
`key` is not used. Can probably be
```
key_type, _ = _parse_identifier(identifier_token)
```

Or maybe you could even not return `key` from `_parse_identifier`

Reply:
This is used on lines 267, 269 and 271.

Reply:
Right, sorry. I even looked for it, but not successfully it turns out 😞